mithril-aggregator/src/signable_builder/immutable_signable_builder.rs

```diff
@@ -0,0 +1,66 @@
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use mithril_common::{
+    digesters::ImmutableDigester,
+    entities::{Beacon, ProtocolMessage, ProtocolMessagePartKey},
+    signable_builder::SignableBuilder,
+    StdResult,
+};
+
+pub struct ImmutableSignableBuilder {
+    immutable_digester: Arc<dyn ImmutableDigester>,
+}
+
+impl ImmutableSignableBuilder {
+    /// Constructor
+    pub fn new(immutable_digester: Arc<dyn ImmutableDigester>) -> Self {
+        Self { immutable_digester }
+    }
+}
+
+#[async_trait]
+impl SignableBuilder<Beacon, ProtocolMessage> for ImmutableSignableBuilder {
+    async fn compute_signable(&self, beacon: Beacon) -> StdResult<ProtocolMessage> {
+        let digest = self.immutable_digester.compute_digest(&beacon).await?;
+        let mut protocol_message = ProtocolMessage::new();
+        protocol_message.set_message_part(ProtocolMessagePartKey::SnapshotDigest, digest);
```

Comment:
IMHO the AVK for the next signing epoch should be added to the protocol message for compatibility and for increasing the chance that there is no epoch gap in the chain :thinking: